### PR TITLE
feat: add user and group pages in Renku 2.0

### LINKED ISSUE #3669: [ShapeUp] User and Group Pages in Renku 2.0

### PR DESCRIPTION
**Pull request by leafty**
Closes #3669.

*Feature branch.*

Renku 2.0 features:
- add user pages
- update group pages
- update project information panel
- update search page with richer search results

/deploy

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,6 +3,33 @@
...
 
+User-Facing Changes
+~~~~~~~~~~~~~~~~~~~
+
+**🌟 New Features**
+
+- **UI**: Renku 2.0: add user pages, update group pages, update project information panel, update the search page (`#3198 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3198>`__)
+- **Search Services**: Search results return details about the creator
+  and project namespace, where before only the respective ids were
+  included
```
Comment (Panaetius) [inline CHANGELOG.rst:14]: nitpick: bit weird line breaking going on here compared to lines above/below? We don't really enforce any line length limits in this file, but should at least keep things consistent.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3,6 +3,33 @@
...
+
+Internal Changes
+~~~~~~~~~~~~~~~~
+
+**🌟 New Features**
+
+- **Search Services**: The search query is now accepted at
+  `/api/search/query` url path and a `/api/search/version` endpoint
+  has been added
```
Comment (Panaetius) [inline CHANGELOG.rst:23]: same here

**Review by Panaetius — CHANGES_REQUESTED** [review]

**Review by Panaetius — APPROVED** [review]

File: CHANGELOG.rst [hunk 3]
```diff
@@ -3,6 +3,32 @@
...
+  where before only the respective ids were included
+
+Internal Changes
+~~~~~~~~~~~~~~~~
+
+**🌟 New Features**
+
+- **Search Services**: The search query is now accepted at ``/api/search/query`` url path
+  and a ``/api/search/version`` endpoint has been added
+- **Data Services**: API changes required to provide user and group pages in Renku 2.0
```
Comment (lokijuhy) [inline CHANGELOG.rst:23]: There's a lot here! So I separated them out into separate line items for easier readability

```suggestion
- **UI**: Renku 2.0: Add user pages that show all projects in the namespace (`#3198 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3198>`__)
- **UI**: Renku 2.0: Extend group pages to show all projects in the namespace (`#3198 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3198>`__)

**✨ Improvements**

- **UI**: Renku 2.0: Provide clickable links between projects and user/group namespace pages on the project page and in search results (`#3198 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3198>`__)
- **Search Services**: Renku 2.0: Show creator name and project namespace in search results,
  where before only the respective ids were included (`#3198 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3198>`__)

Internal Changes
~~~~~~~~~~~~~~~~

**Improvements**

- **Search Services**: The search query is now accepted at ``/api/search/query`` url path
  and a ``/api/search/version`` endpoint has been added
- **Data Services**: Change API to provide user and group pages in Renku 2.0
```

**Review by lokijuhy — CHANGES_REQUESTED** [review]
If accepting these changes requires otherwise unnecessary re-running of tests, let me know and I'll approve and make these changes later.

**Review by lokijuhy — APPROVED** [review]

**Pull request merged**